uk_bin_collection/uk_bin_collection/councils/IsleOfWightCouncil.py

```
@@ -0,0 +1,454 @@
+                    "strong", string=re.compile(r"Collection Day:")
+                )
+                if not collection_day_el:
+                    raise ValueError("Could not find collection day in page")
+                collection_day = (
+                    collection_day_el.parent.get_text()
+                    .replace("Collection Day:", "")
+                    .strip()
+                )
+
+                os.makedirs(PDF_CACHE_DIR, exist_ok=True)
+                cache_key = hashlib.md5(
```

Ruff (0.15.12)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)